### ISSUE TITLE
Annotations type change dynamically 

### ISSUE THREAD

**Issue by Eugene88** [susielu/d3-annotation#46]
Hi,
I have a problem with  dynamically changing annotation type. I tring change annotation type when I click on note.

var makeAnnotation= d3.annotation().annotations(annotations0).type(d3.annotationCallout).on('noteclick', onClick)

function onClick(annotation) 
{
   annotation.color = "#000000"; - it works
   annotation.type = d3.annotationLabel;  - it doesn't work
   makeAnnotation.update();
}

I can change type for all items from annotations0, but I need change a single annotation type.
It works:
makeAnnotation.type(d3.annotationLabel)
d3.select("#groupId0").select("g.annotation-group").call(makeAnnotation)

How can I  change a single annotation type?

Thanks.

**Comment by susielu** [comment]
I should post a better example of this, if you re-call the function on the selection it should update the type. So assuming you were doing something like this to create the annotations
```js
d3.select("svg")
  .append("g")
  .attr("class", "annotation-group")
  .call(makeAnnotations)
```
You would then change you onClick to something like this : 
```js
function onClick(annotation) {
  annotation.color = "#000000"; 
  annotation.type = d3.annotationLabel;
  // makeAnnotation.update(); <- this does reset type
  d3.select(".annotation-group") //<- this should
    .call(makeAnnotations)
}

**Comment by Eugene88** [comment]
Thanks. It works.
But I have a problem with d3.annotationBadge. Badge was created and not removed after change type
![annotation](https://user-images.githubusercontent.com/605347/37231858-c3c417d2-23fd-11e8-9c80-31c540efc9f5.gif)
I switch types [d3.annotationLabel, d3.annotationCallout, d3.annotationCalloutElbow, d3.annotationBadge]

**Comment by Eugene88** [comment]
And type doesn't change to d3.annotationLabel too  (visualisation not  changed)

**Comment by susielu** [comment]
I believe this version fixes this `2.2.1` let me know if you have any more issues with it.

**Comment by susielu** [comment]
Oh no there was a bug in that version `2.2.2` should work.
